flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -305,4 +308,52 @@ public void invoke(Integer value) throws Exception {
 			}
 		}
 	}
+
+	@Test(expected = IllegalStateException.class)
+	public void testDisabledAutoUIDGenerator() {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.getConfig().disableAutoGeneratedUIDs();
+
+		env.fromElements(1, 2, 3).map(new NoOpIntMap()).addSink(new DiscardingSink<>());
+		StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
+
+		Assert.fail("Accepted job graph without manually set UIDs");
+	}
+
+	@Test
+	public void testDisabledAutoUIDWithManuallySetId() {
```

Comment:
Shouldn't all `DisabledAutoUID*` tests fail currently? Each of the test have operators without uids assigned (e.g. `.map(new NoOpIntMap())`, `.addSink(new DiscardingSink<>())`),

Reply:
No, again because of operator chaining. That’s why the third test explicitly disables chaining in one place to check that condition.